src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.csproj

```diff
@@ -28,5 +28,6 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Azure.Identity" Version="$(AzureIdentityVersion)" />
+    <PackageReference Include="Microsoft.Identity.Client.Broker" Version="4.64.0" />
```

Comment:
Picking up from JRahnama's comments - I imagine you'll probably have to conditionally reference Microsoft.Identity.Client.Broker on net8.0 and Microsoft.Identity.Client.Desktop on net462/net6.0.

Just for awareness, M.I.C.D will drag in Microsoft.Web.WebView2 - an 8MB package.

Reply:
I don't see a better or more straightforward option than using an environment variable to determine if M.I.C.B is needed by the client. 

I was hoping to find a way to read this from the client's config or JSON file, but that seems more challenging.

Reply:
The [surrounding documentation](https://learn.microsoft.com/en-gb/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam#enabling-wam) says:

> For migration purposes, and if you have a .NET 6, .NET Core, or a .NET Standard application that needs to use both WAM and the [embedded browser](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-net-web-browsers#embedded-vs-system-web-ui), you will also need to use the [Microsoft.Identity.Client.Desktop](https://www.nuget.org/packages/Microsoft.Identity.Client.Desktop/) package.

It's worth checking, but I think this means we can use MIC.Broker on .NET 8.0, and MIC.Desktop on .NET Framework or .NET 6.0. At that point, I'd expect it to be nothing more than an MSBuild `Condition` attribute in the netcore project file.

It's worth keeping in mind that .NET 6.0 is leaving support on 12th November, so if SqlClient v6.0 drops support at the same time this question might just be academic.

Reply:
I was considering adding something like `IsWamRequested`, which could be sourced from a JSON file or application config within the client application. If this flag is set, the relevant setup and configuration could be enabled or applied dynamically. This was just an idea that came to mind... Just thinking out loud here  😆